### ISSUE TITLE
Unicode flag

### ISSUE THREAD

**Issue by jengu** [realm/SwiftLint#3210]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Unable to add global flag for custom rule regex. I want to handle cases which include russian letters, thus I have to use unicode flag for regex. Is it supported?

### Environment

* SwiftLint version - 0.38.2 (I've tried updating to last version, no changes)
* Installation method used - CocoaPods
* Paste your configuration file:

```yml
custom_rules:
  non_breaking_space:
      regex: 'tr\("{1}.*\b(в)\b( )'
      message: "Non-breaking space is required for prepositions and conjunctions"
      severity: warning
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations) - No
* Which Xcode version are you using - 11.4.1

```swift
// wrongly triggering case
return tr("")
```
